src/main/java/org/jabref/logic/ai/models/EmbeddingModel.java

```diff
@@ -0,0 +1,104 @@
+package org.jabref.logic.ai.models;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
+
+import org.jabref.logic.l10n.Localization;
+import org.jabref.preferences.AiPreferences;
+
+import ai.djl.MalformedModelException;
+import ai.djl.huggingface.translator.TextEmbeddingTranslatorFactory;
+import ai.djl.repository.zoo.Criteria;
+import ai.djl.repository.zoo.ModelNotFoundException;
+import ai.djl.training.util.ProgressBar;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.output.Response;
+
+/**
+ * Wrapper around langchain4j embedding model.
+ * <p>
+ * This class listens to preferences changes.
+ */
+public class EmbeddingModel implements dev.langchain4j.model.embedding.EmbeddingModel, AutoCloseable {
+    private final AiPreferences aiPreferences;
+
+    private final ExecutorService executorService = Executors.newCachedThreadPool(
+            new ThreadFactoryBuilder().setNameFormat("ai-embedding-pool-%d").build()
+    );
+
+    private final ObjectProperty<Optional<DeepJavaEmbeddingModel>> predictorProperty = new SimpleObjectProperty<>(Optional.empty());
+
+    public EmbeddingModel(AiPreferences aiPreferences) {
+        this.aiPreferences = aiPreferences;
+        rebuild();
+        setupListeningToPreferencesChanges();
+    }
+
+    private void rebuild() {
+        if (!aiPreferences.getEnableChatWithFiles()) {
+            predictorProperty.set(Optional.empty());
+            return;
+        }
+
+        String modelUrl = "djl://ai.djl.huggingface.pytorch/sentence-transformers/" + aiPreferences.getEmbeddingModel().getLabel();
```

Comment:
Non-trivial constants should be made class constants ( `private static final ..`). Reason: More easy identifiable. Maybe even possible to configure it. Or "just" change it. We can search for variable names "URL_BASE" for instance. (Although we currently do not have any strict method for that in the source code).